pkg/controller/servicemesh/webhooks/validation/controlplane_upgrade_v1_0.go

```diff
@@ -0,0 +1,117 @@
+package validation
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	pkgerrors "github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	maistrav1 "github.com/maistra/istio-operator/pkg/apis/maistra/v1"
+)
+
+var (
+
+	// These are unsupported in v1.1
+	unsupportedOldResourcesV1_1 = []schema.GroupVersionKind{
```

Comment:
I'm not sure we should be rejecting the upgrade if these resources are present. By doing so, we're forcing users to first delete these resources and then update the SMCP.  But users usually update multiple resources in a single kubectl apply command. Plus, doesn't the operator itself create some of these resources when you create the SMCP? Does this mean they have to delete them manually even though they never created them?

Reply:
Good question.  First, controller managed resources are filtered from the check, so those will not prevent an upgrade.  Regarding user created CRs, I did not check to see if there was an upgrade path.  These were deprecated in Istio 1.2 and removed in Istio 1.3, and since we're moving from Istio 1.1 to 1.4, it's likely there isn't anything in Istio 1.1 that supports migration to a new resource.  Is it possible to return a warning from validation?